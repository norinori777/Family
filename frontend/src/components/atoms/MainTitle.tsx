import { theme } from '../../util/theme/types'
import { getTextTheme } from '../../util/theme/theme'
import React from 'react'

export interface MainTitleProps {
  title: string
  theme: theme
}

export const MainTitle = (props: MainTitleProps) => {
  const textTheme = getTextTheme(props.theme)
  return (
    <div className="w-72">
      <h1 className={`${textTheme} text-2xl font-bold`}>{props.title}</h1>
    </div>
  )
}
