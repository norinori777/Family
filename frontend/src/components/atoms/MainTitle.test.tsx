import React from 'react'
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { MainTitle, MainTitleProps } from './MainTitle'

const props: MainTitleProps = {
  title: 'タイトル',
  theme: 'primary',
}

describe('MainTitle', () => {
  test('タイトルの表示', () => {
    const { container } = render(<MainTitle {...props} />)
    const title = screen.getByRole('heading', { level: 1 })
    expect(container).toMatchSnapshot()
    expect(title).toHaveTextContent('タイトル')
  })
  test('primaryテーマの確認', () => {
    render(<MainTitle {...props} />)
    const title = screen.getByRole('heading', { level: 1 })
    expect(title).toHaveClass('text-primary-default')
  })
})
