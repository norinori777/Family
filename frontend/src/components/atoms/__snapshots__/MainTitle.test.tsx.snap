// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainTitle タイトルの表示 1`] = `
<div>
  <div
    class="w-72"
  >
    <h1
      class="text-primary-default text-2xl font-bold"
    >
      タイトル
    </h1>
  </div>
</div>
`;

exports[`タイトルの表示 1`] = `
<div>
  <div
    class="w-72"
  >
    <h1
      class="text-primary-default text-2xl font-bold"
    >
      タイトル
    </h1>
  </div>
</div>
`;
